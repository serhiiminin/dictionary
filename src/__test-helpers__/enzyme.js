import React from 'react';
import { shallow, mount } from 'enzyme';
import { ThemeProvider } from 'styled-components';

const shallowWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()
  return shallow(tree, { context })
}

const mountWithTheme = (tree, theme) => {
  const context = shallow(<ThemeProvider theme={theme} />)
    .instance()
    .getChildContext()

  return mount(tree, {
    context,
    childContextTypes: ThemeProvider.childContextTypes,
  })
}

export { shallowWithTheme, mountWithTheme }
