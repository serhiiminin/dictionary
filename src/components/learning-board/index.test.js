import React from 'react';
import { shallow } from 'enzyme';
import LearningBoard from '.';
import muiTheme from '../../root/mui-theme';

describe('LearningBoard', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <LearningBoard
        theme={muiTheme}
        onCheckAnswer={() => {}}
        onGiveAHint={() => {}}
        onKnownWord={() => {}}
        onForgottenWord={() => {}}
        onOptionChange={() => {}}
      >
        anything
      </LearningBoard>
    );
  });
  test('render', () => {
    expect(wrapper).toMatchSnapshot();
  });
  test('loading', () => {
    wrapper.setProps({ loading: true });
    expect(wrapper).toMatchSnapshot();
  });
  test('times learnt is more then 1', () => {
    wrapper.setProps({ timesLearnt: 2 });
    expect(wrapper).toMatchSnapshot();
  });
});
