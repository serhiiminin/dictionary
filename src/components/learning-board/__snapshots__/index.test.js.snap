// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LearningBoard loading 1`] = `
<div>
  <TextFieldLoading
    disabled={true}
    label="Your option"
    loading={true}
    onChange={[Function]}
    value={null}
  />
  <h3 />
  <h5 />
  <WithTheme(ControlsSeparator)
    align="center"
  >
    <ButtonControl
      disabled={true}
      onClick={[Function]}
      title="Submit my answer"
    >
      <pure(DoneIcon) />
    </ButtonControl>
    <ButtonControl
      disabled={true}
      onClick={[Function]}
      title="Give me a hint"
    >
      <pure(RemoveRedEyeIcon) />
    </ButtonControl>
    <ButtonControl
      disabled={true}
      onClick={[Function]}
      title="I know this word"
    >
      <pure(DoneAllIcon) />
    </ButtonControl>
    <ButtonControl
      disabled={true}
      onClick={[Function]}
      title="I forgot this word, show me the translation"
    >
      <pure(ErrorOutlineIcon) />
    </ButtonControl>
  </WithTheme(ControlsSeparator)>
</div>
`;

exports[`LearningBoard render 1`] = `
<div>
  <TextFieldLoading
    disabled={null}
    label="Your option"
    loading={null}
    onChange={[Function]}
    value={null}
  />
  <h3 />
  <h5 />
  <WithTheme(ControlsSeparator)
    align="center"
  >
    <ButtonControl
      disabled={null}
      onClick={[Function]}
      title="Submit my answer"
    >
      <pure(DoneIcon) />
    </ButtonControl>
    <ButtonControl
      disabled={null}
      onClick={[Function]}
      title="Give me a hint"
    >
      <pure(RemoveRedEyeIcon) />
    </ButtonControl>
    <ButtonControl
      disabled={null}
      onClick={[Function]}
      title="I know this word"
    >
      <pure(DoneAllIcon) />
    </ButtonControl>
    <ButtonControl
      disabled={null}
      onClick={[Function]}
      title="I forgot this word, show me the translation"
    >
      <pure(ErrorOutlineIcon) />
    </ButtonControl>
  </WithTheme(ControlsSeparator)>
</div>
`;

exports[`LearningBoard times learnt is more then 1 1`] = `
<div>
  <TextFieldLoading
    disabled={null}
    label="Your option"
    loading={null}
    onChange={[Function]}
    value={null}
  />
  <h3 />
  <h5>
    Learnt 2 times before
  </h5>
  <WithTheme(ControlsSeparator)
    align="center"
  >
    <ButtonControl
      disabled={null}
      onClick={[Function]}
      title="Submit my answer"
    >
      <pure(DoneIcon) />
    </ButtonControl>
    <ButtonControl
      disabled={null}
      onClick={[Function]}
      title="Give me a hint"
    >
      <pure(RemoveRedEyeIcon) />
    </ButtonControl>
    <ButtonControl
      disabled={null}
      onClick={[Function]}
      title="I know this word"
    >
      <pure(DoneAllIcon) />
    </ButtonControl>
    <ButtonControl
      disabled={null}
      onClick={[Function]}
      title="I forgot this word, show me the translation"
    >
      <pure(ErrorOutlineIcon) />
    </ButtonControl>
  </WithTheme(ControlsSeparator)>
</div>
`;
